OCPBUGS#5193: Added descriptive commands in opm CLI reference section. (#57337)
openshift/openshift-docs: modules/opm-cli-ref-generate.adoc

//Module included in the following assemblies:
//
//*cli_reference/opm/cli-opm-ref.adoc

[id="opm-cli-ref-generate_{Context}"]
= generate

Generate various artifacts for declarative config indexes.

.Command syntax
[source,terminal]
----
$ opm generate [command]
----

.`generate` flags
[options="header",cols="1,3"]
|===
|Flags |Description

|`-h`, `--help`
|Help for generate.

|===

.Command syntax
[source,terminal]
----
$ opm generate <subcommand> [<flags>]
----

.`generate` subcommands
[options="header",cols="1,3"]
|===
|Subcommand |Description

|`dockerfile`
|Generate a Dockerfile for a declarative config index.

|===

[id="opm-cli-ref-generate-dockerfile_{context}"]
== dockerfile

Generate a Dockerfile for a declarative config index.

[IMPORTANT]
====

> This command creates a Dockerfile in the same directory as the <dcRootDir> (named <dcDirName>.Dockerfile) that can be used to build the index. If a Dockerfile with the same name already exists, this command fails.

Author: Addressed.

> When specifying extra labels, note that if duplicate keys exist, only the last value of each duplicate key will be added to the generated Dockerfile.

Reviewer: ```suggestion
When specifying extra labels, if duplicate keys exist, only the last value of each duplicate key gets added to the generated Dockerfile.
```

====

.Command syntax
[source,terminal]
----
$ opm generate dockerfile <dcRootDir> [flags]
----

.`generate dockerfile` flags
[options="header",cols="1,3"]
|===
|Flag |Description

|`-i,` `--binary-image` (string)

> |Image in which to build catalog. Default: "quay.io/operator-framework/opm:latest".

Reviewer: ```suggestion
|Image in which to build catalog. The default value is `quay.io/operator-framework/opm:latest`.
```

|`-l`, `--extra-labels` (string)

> |Extra labels to include in the generated Dockerfile. Labels should be of the form `key=value`.

Author: Done.

> |Extra labels to include in the generated Dockerfile. Labels should be of the form 'key=value'.

Reviewer: ```suggestion
|Extra labels to include in the generated Dockerfile. Labels should be of the form `key=value`.
```
